### ISSUE TITLE
loss_weight is randomly re-initialized at each step

### ISSUE THREAD

**Issue by pmorerio** [ZangXuan/HiMol#3]
https://github.com/ZangXuan/HiMol/blob/ddf4973b823f0e7907230abf61ac9ce96d7e476e/decoder.py#L158

Hi, thank you very much for sharing the code and for the nice work.

According to the paper

![image](https://github.com/ZangXuan/HiMol/assets/6783835/876e251f-e992-4c74-85f1-35b93fae51ad)

From the code however it looks like the loss weight is randomly initialized at each forward pass. 
I guess you should define 
```
self.loss_weight = create_var(torch.rand(5),self.device, requires_grad=True)
```
in the `__init__` method and then use that variable in the forward pass. Otherwise you are using different random weights at each step.
